auslib/web/admin/views/rules.py

```diff
@@ -117,10 +118,13 @@ def _post(self, id_or_alias, transaction, changed_by):
 
         what, mapping_values, fallback_mapping_values = process_rule_form(connexion.request.get_json())
 
-        if what.get('mapping', None) is not None and len(mapping_values) != 1:
+        if 'mapping' in what and not what.get('mapping', None):
+            return problem(400, 'Bad Request', 'mapping value cannot be set to null/empty')
+
+        elif what.get('mapping', None) and len(mapping_values) != 1:
             return problem(400, 'Bad Request', 'Invalid mapping value. No release name found in DB')
 
-        if what.get('fallbackMapping', None) is not None and len(fallback_mapping_values) != 1:
+        elif what.get('fallbackMapping') is not None and len(fallback_mapping_values) != 1:
```

Comment:
Update 2: this [line](https://github.com/mozilla/balrog/blob/master/auslib/web/admin/views/rules.py#L30) is irrelevant because [here](https://github.com/mozilla/balrog/blob/master/auslib/web/admin/views/rules.py#L36) we replace all blank field values to None , hence while constructing the 'mapping_values' list on these lines [1](https://github.com/mozilla/balrog/blob/master/auslib/web/admin/views/rules.py#L44), [2](https://github.com/mozilla/balrog/blob/master/auslib/web/admin/views/rules.py#L45), x i.e. key will never match a blank value from rule_form_dict.get("mapping", None) since it has been set to None if it was blank initially and hence the list will be empty. So no issues. If mapping value was indeed provided and list was empty then 400 response will be sent.